### PR TITLE
add latest anomaly notebook in zouwu

### PR DESCRIPTION
**Pull request by EN1AC13**
add latest zoo API in anomaly notebook
add plot function 
fix data pre-processing

### CODE REVIEW

File: pyzoo/zoo/zouwu/use-case/AIOps/extract_data.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+fi
+
+FILENAME="./m_1932.csv"
+if [ -f "$FILENAME" ]
+then
+   echo "$FILENAME already exists."
+   exit
+fi
+
+echo "Extrat m_1932"
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/extract_data.sh:20]: extract

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 2]
```diff
@@ -0,0 +1,1485 @@
...
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Now we need to do data cleaning and preprocessing on the raw data. Note that this part could vary for different dataset. \n",
+    "\n",
+    "For the machine_usage data, we are using the processing contains 2 parts:\n",
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:423]: the pre-processing contains 2 parts

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 3]
```diff
@@ -0,0 +1,1485 @@
...
+   "outputs": [],
+   "source": [
+    "df_1932[\"ts_min\"] = df_1932[\"time_step\"]//60"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Count the average of cpu_usage in the same time and draw the line chart."
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:452]: calculate

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 4]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Last, we generate model input samples by sliding window along time axis. Since univariate and multivariate forecasting uses different input data shapes, we prepare data for each type of forecasting respectively."
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:982]: remove the second sentence

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 5]
```diff
@@ -0,0 +1,1485 @@
...
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Now we need to do data cleaning and preprocessing on the raw data. Note that this part could vary for different dataset. \n",
+    "\n",
+    "For the machine_usage data, we are using the processing contains 2 parts:\n",
+    "1. Change the time unit, and then count the average of `cpu_usage` in the same time\n",
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:424]: 1. Change the time unit from seconds to minutes, which is calculated by averaging `cpu_usage` within the minutes.

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 6]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "First, we initialize a mtnet_forecaster according to input data shape. Specifcally, look_back should equal (lb_long_steps+1)*lb_long_stepsize .Details refer to zouwu docs."
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:1066]: `(lb_long_steps+1)*lb_long_stepsize`
add link for "zouwu doc"

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 7]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "MTNet needs to preprocess the X into another format, so we call MTNetForecaster.preprocess_input on train_x and test_x."
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:1107]: ` MTNetForecaster.preprocess_input`

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 8]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Use the model for prediction and inverse the scaling of the prediction results"
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:1212]: keep one annotation between this and the one below.

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 9]
```diff
@@ -0,0 +1,1485 @@
...
+   "source": [
+    "y_pred_val = mtnet_forecaster.predict(x_val_mtnet)\n",
+    "y_pred_test = mtnet_forecaster.predict(x_test_mtnet)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Since we have used standard scaler to scale the input data (including the target values), we need to inverse the scaling on the predicted values too. "
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:1233]: repeat explanation of "inverse the scaling..."

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 10]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "pycharm": {
+     "name": "#%% md\n"
+    }
+   },
+   "source": [
+    "Use anomaly detector interface in zoo.zouwu.model.anomaly"
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:1304]: maybe this sentence could be removed

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb [hunk 11]
```diff
@@ -0,0 +1,1485 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 16,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "resampled_df.index = pd.to_datetime(resampled_df.index, unit='m', origin=pd.Timestamp('2018-01-01'))\n",
+    "full_idx = pd.date_range(start=resampled_df.index.min(), end=resampled_df.index.max(),freq='T')\n",
+    "resampled_df = resampled_df.reindex(full_idx)"
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb:519]: needs explanation

File: pyzoo/zoo/zouwu/use-case/AIOps/extract_data.sh [hunk 12]
```diff
@@ -0,0 +1,23 @@
...
+
+FILENAME="./m_1932.csv"
+if [ -f "$FILENAME" ]
+then
+   echo "$FILENAME already exists."
+   exit
+fi
+
+echo "Extract m_1932"
+grep m_1932 machine_usage.csv > m_1932.csv
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/use-case/AIOps/extract_data.sh:21]: check whether machine_usage.csv exists. If not, give a hint to run `get_data.sh` first.

**Comment by EN1AC13** [comment]
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration-Part4/30/console passed

**Comment by yushan111** [comment]
remove comment for other app tests @EN1AC13

File: pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb (hunk 2 above)
Comment (yushan111) [reply pyzoo/zoo/zouwu/use-case/AIOps/AIOps_anomaly_detect.ipynb]: remove we are using. @EN1AC13

**Comment by EN1AC13** [comment]
http://10.239.47.210:18888/view/ZOO-PR/job/ZOO-PR-Pip-Apps-Integration-Part4/35/console passed

**Pull request merged**